homeassistant/components/media_player/significant_change.py

```diff
@@ -0,0 +1,71 @@
+"""Helper to test significant Media Player state changes."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.significant_change import (
+    check_absolute_change,
+    check_valid_float,
+)
+
+from . import (
+    ATTR_ENTITY_PICTURE_LOCAL,
+    ATTR_GROUP_MEMBERS,
+    ATTR_MEDIA_POSITION,
+    ATTR_MEDIA_POSITION_UPDATED_AT,
+    ATTR_MEDIA_VOLUME_LEVEL,
+    ATTR_TO_PROPERTY,
+)
+
+INSIGNIFICANT_ATTRIBUTES: set[str] = {
+    ATTR_MEDIA_POSITION,
+    ATTR_MEDIA_POSITION_UPDATED_AT,
+}
+
+SIGNIFICANT_ATTRIBUTES: set[str] = {
+    ATTR_ENTITY_PICTURE_LOCAL,
+    ATTR_GROUP_MEMBERS,
+    *ATTR_TO_PROPERTY,
+}
+
+
+@callback
+def async_check_significant_change(
+    hass: HomeAssistant,
+    old_state: str,
+    old_attrs: dict,
+    new_state: str,
+    new_attrs: dict,
+    **kwargs: Any,
+) -> bool | None:
+    """Test if state significantly changed."""
+    if old_state != new_state:
+        return True
+
+    old_attrs_s = set(old_attrs.items())
+    new_attrs_s = set(new_attrs.items())
+    changed_attrs: set[str] = {item[0] for item in old_attrs_s ^ new_attrs_s}
+
+    for attr_name in changed_attrs:
+        if attr_name not in SIGNIFICANT_ATTRIBUTES - INSIGNIFICANT_ATTRIBUTES:
```

Comment:
Taking the difference between two constants gives another constant. Define that constant at the module level.

Reply:
```python
SIGNIFICANT_ATTRIBUTES: set[str] = {
    ATTR_ENTITY_PICTURE_LOCAL,
    ATTR_GROUP_MEMBERS,
    *ATTR_TO_PROPERTY,
} ^ {ATTR_MEDIA_POSITION, ATTR_MEDIA_POSITION_UPDATED_AT}
```

would this be ok to just have `SIGNIFICANT_ATTRIBUTES`? if yes, i'll create a PR

Reply:
or maybe this one 🤔 
```python
INSIGNIFICANT_ATTRIBUTES: set[str] = {
    ATTR_MEDIA_POSITION,
    ATTR_MEDIA_POSITION_UPDATED_AT,
}

SIGNIFICANT_ATTRIBUTES: set[str] = {
    ATTR_ENTITY_PICTURE_LOCAL,
    ATTR_GROUP_MEMBERS,
    *ATTR_TO_PROPERTY,
} ^ INSIGNIFICANT_ATTRIBUTES
```

Reply:
Yes, but note that the difference is not the same as the symmetric difference of two sets.

Reply:
uhhh ... good to known 👍 

```python
Python 3.11.4 (main, Jun  7 2023, 18:32:58) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {1,2,3,4,5,6}
>>> b = {3,5,7}
>>> a - b
{1, 2, 4, 6}
>>> a ^ b
{1, 2, 4, 6, 7}
```

than it should be like this, right?

```
INSIGNIFICANT_ATTRIBUTES: set[str] = {
    ATTR_MEDIA_POSITION,
    ATTR_MEDIA_POSITION_UPDATED_AT,
}

SIGNIFICANT_ATTRIBUTES: set[str] = {
    ATTR_ENTITY_PICTURE_LOCAL,
    ATTR_GROUP_MEMBERS,
    *ATTR_TO_PROPERTY,
} - INSIGNIFICANT_ATTRIBUTES
```

Reply:
Yes. 👍

Reply:
- #106727 ✅